src/pynwb/io/file.py

```diff
@@ -156,28 +182,54 @@ def scratch(self, builder, manager):
 
     @ObjectMapper.constructor_arg('session_start_time')
     def dateconversion(self, builder, manager):
+        """Set the constructor arg for 'session_start_time' to a datetime object.
+
+        Used when constructing the NWBFile container from a written file.
+
+        Dates are read into builders as strings and are parsed into datetime objects
+        for user convenience and consistency with how they are written.
+        """
         datestr = builder.get('session_start_time').data
         date = dateutil_parse(datestr)
         return date
 
     @ObjectMapper.constructor_arg('timestamps_reference_time')
     def dateconversion_trt(self, builder, manager):
+        """Set the constructor arg for 'timestamps_reference_time' to a datetime object.
+
+        Used when constructing the NWBFile container from a written file.
+
+        Dates are read into builders as strings and are parsed into datetime objects
+        for user convenience and consistency with how they are written.
+        """
         datestr = builder.get('timestamps_reference_time').data
         date = dateutil_parse(datestr)
         return date
 
     @ObjectMapper.constructor_arg('file_create_date')
     def dateconversion_list(self, builder, manager):
+        """Set the constructor arg for 'file_create_date' to a datetime object.
+
+        Used when constructing the NWBFile container from a written file.
+
+        Dates are read into builders as strings and are parsed into datetime objects
+        for user convenience and consistency with how they are written.
+        """
         datestr = builder.get('file_create_date').data
         dates = list(map(dateutil_parse, datestr))
         return dates
 
-    @ObjectMapper.constructor_arg('file_name')
```

Comment:
just wanted to clarify why this constructor was removed? Otherwise, looks good to me!

Reply:
Whoops I forgot to mention that. This maps a constructor argument that doesn't exist. It must have been a legacy thing.